main.tf

```
@@ -20,4 +21,66 @@ resource "opentelekomcloud_obs_bucket" "bucket" {
     kms_key_id = opentelekomcloud_kms_key_v1.bucket_kms_key.id
   }
   tags = var.tags
+
+  dynamic "worm_policy" {
+    for_each = var.enable_worm_policy ? [1] : []
```

Maybe to reduce the number of variables:

```suggestion
    for_each = var.worm_policy_years ? [1] : []
```